src/common/vulkan/util.h

```diff
@@ -74,10 +74,141 @@ VkShaderModule CompileAndCreateFragmentShader(std::string_view source_code);
+template<> struct VkObjectTypeMap<VkBufferView				> { using type = VkBufferView				; static constexpr VkObjectType value = VK_OBJECT_TYPE_BUFFER_VIEW; };
+template<> struct VkObjectTypeMap<VkImageView				> { using type = VkImageView				; static constexpr VkObjectType value = VK_OBJECT_TYPE_IMAGE_VIEW; };
+template<> struct VkObjectTypeMap<VkShaderModule			> { using type = VkShaderModule				; static constexpr VkObjectType value = VK_OBJECT_TYPE_SHADER_MODULE; };
+template<> struct VkObjectTypeMap<VkPipelineCache			> { using type = VkPipelineCache			; static constexpr VkObjectType value = VK_OBJECT_TYPE_PIPELINE_CACHE; };
+template<> struct VkObjectTypeMap<VkPipelineLayout			> { using type = VkPipelineLayout			; static constexpr VkObjectType value = VK_OBJECT_TYPE_PIPELINE_LAYOUT; };
+template<> struct VkObjectTypeMap<VkRenderPass				> { using type = VkRenderPass				; static constexpr VkObjectType value = VK_OBJECT_TYPE_RENDER_PASS; };
+template<> struct VkObjectTypeMap<VkPipeline				> { using type = VkPipeline					; static constexpr VkObjectType value = VK_OBJECT_TYPE_PIPELINE; };
+template<> struct VkObjectTypeMap<VkDescriptorSetLayout		> { using type = VkDescriptorSetLayout		; static constexpr VkObjectType value = VK_OBJECT_TYPE_DESCRIPTOR_SET_LAYOUT; };
+template<> struct VkObjectTypeMap<VkSampler					> { using type = VkSampler					; static constexpr VkObjectType value = VK_OBJECT_TYPE_SAMPLER; };
+template<> struct VkObjectTypeMap<VkDescriptorPool			> { using type = VkDescriptorPool			; static constexpr VkObjectType value = VK_OBJECT_TYPE_DESCRIPTOR_POOL; };
+template<> struct VkObjectTypeMap<VkDescriptorSet			> { using type = VkDescriptorSet			; static constexpr VkObjectType value = VK_OBJECT_TYPE_DESCRIPTOR_SET; };
+template<> struct VkObjectTypeMap<VkFramebuffer				> { using type = VkFramebuffer				; static constexpr VkObjectType value = VK_OBJECT_TYPE_FRAMEBUFFER; };
+template<> struct VkObjectTypeMap<VkCommandPool				> { using type = VkCommandPool				; static constexpr VkObjectType value = VK_OBJECT_TYPE_COMMAND_POOL; };
+template<> struct VkObjectTypeMap<VkDescriptorUpdateTemplate> { using type = VkDescriptorUpdateTemplate	; static constexpr VkObjectType value = VK_OBJECT_TYPE_DESCRIPTOR_UPDATE_TEMPLATE; };
+template<> struct VkObjectTypeMap<VkSurfaceKHR				> { using type = VkSurfaceKHR				; static constexpr VkObjectType value = VK_OBJECT_TYPE_SURFACE_KHR; };
+template<> struct VkObjectTypeMap<VkSwapchainKHR			> { using type = VkSwapchainKHR				; static constexpr VkObjectType value = VK_OBJECT_TYPE_SWAPCHAIN_KHR; };
+template<> struct VkObjectTypeMap<VkDebugUtilsMessengerEXT	> { using type = VkDebugUtilsMessengerEXT	; static constexpr VkObjectType value = VK_OBJECT_TYPE_DEBUG_UTILS_MESSENGER_EXT; };
+// clang-format on
+
+void SetObjectName(VkDevice device, void* object_handle, VkObjectType object_type, const char* object_name);
```

Comment:
Any reason why these can't be varargs? (avoids the heap allocation for strings, which can be slow in debug builds)